### PR TITLE
🐛 [TK-2026] Fix race condition in `Workspace` and `Module` outputs

### PR DESCRIPTION
**Pull request by arybolovlev**
### Description

This PR addresses intermittent race condition issues in the `Workspace` and `Module` controllers when they create or update outputs. The issue happens when the ongoing Run is finished while the operator is reconciling `Workspace` or `Module`. In this case, method `reconcileOutputs` has outdated information about `CurrentStateVersion` and either does not create related `ConfigMap` or `Secret` or does not update them with the recent changes. If that happens, the outputs will remain outdated until the next Run happens.

This PR adds an additional API call to get an updated TFC Workspace item when `Run` finishes between the time when a new reconciliation starts and synchronizes outputs. The method `reconcileOutputs` now is the one responsible for updating the `Status` of the relevant Kubernetes object.

### CODE REVIEW

File: controllers/suite_test.go [hunk 1]
```diff
@@ -91,10 +94,15 @@ var _ = BeforeSuite(func() {
...
-	syncPeriod := 30 * time.Second
 	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
 		Scheme:     scheme.Scheme,
 		SyncPeriod: &syncPeriod,
+		Controller: v1alpha1.ControllerConfigurationSpec{
+			GroupKindConcurrency: map[string]int{
+				"Workspace.app.terraform.io": 5,
+				"Module.app.terraform.io":    5,
+			},
+		},
```
Comment (arybolovlev) [inline controllers/suite_test.go:105]: This is not really related to this issue, but this will be useful for further improvements to make tests parallel. 🙃

File: controllers/module_controller_test.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ var _ = Describe("Workspace controller", Ordered, func() {
 
 	BeforeAll(func() {
 		// Set default Eventually timers
-		SetDefaultEventuallyTimeout(120 * time.Second)
+		SetDefaultEventuallyTimeout(syncPeriod * 4)
```
Comment (arybolovlev) [inline controllers/module_controller_test.go:26]: This is also not relevant to the issue, just an improvement I made while working on tests.

**Comment by arybolovlev** [comment]
One thing worth mentioning here. The controller `Module` has different methods to update the instance state depending on the current FSM step. With this update, it looks like it can be simplified and only one method can be left. The instance state will be updated on the appropriate step and the method will just do the action. That will be done later to not mix different things in one PR.

File: controllers/suite_test.go [hunk 3]
```diff
@@ -80,6 +83,12 @@ var _ = BeforeSuite(func() {
 
 	//+kubebuilder:scaffold:scheme
 
+	if organization == "" {
+		Fail("Environment variable TFC_ORG is either not set or empty but required")
```
Comment (alexsomesan) [inline controllers/suite_test.go:87]:
```suggestion
		Fail("Environment variable TFC_ORG is required, but either not set or empty")
```

File: controllers/suite_test.go [hunk 4]
```diff
@@ -80,6 +83,12 @@ var _ = BeforeSuite(func() {
...
+	}
+	if terraformToken == "" {
+		Fail("Environment variable TFC_TOKEN is either not set or empty but required")
```
Comment (alexsomesan) [inline controllers/suite_test.go:90]:
```suggestion
		Fail("Environment variable TFC_TOKEN is required, but either not set or empty")
```

**Review by alexsomesan — APPROVED** [review]
Looks good. 
Small rephrasing suggestions on some error messages.
